### DISCUSSION TITLE
renderJson in 1.4.5

### DISCUSSION THREAD

**Discussion by adamcameron** [wheels-dev/wheels#1196] [General]
G'day:
I'm pretty sure I know the answer to this, but as there's no `renderJson` method or anything that looks like that in 1.4.5, is the CFWheels-idiomatic way of returning JSON from an end point to simply `renderText(myJsonHere)`, and explicitly use `<cfheader name="Content-Type" value="application/json">` to set the content type?

As it's such a fundamental thing to want to do, I'm a bit surprised to not find a pre-baked solution for it, so wonder if I'm just being a div and missing something obvious?

-- 
Adam

**Comment by neokoenig** [comment]
For 1.4x: In your controllers `init()` function, you can add 
`provides("json");`

Then in your action
```
foo(){
  // This can be set via `/foo?format=json` in the URL, or via `/foo.json` but I find it works better if you explicitedly set it, then `/foo` will always return json
  params.format = 'json';

  mystuff = { foobar: 'foo' }
  renderWith(mystuff)
}
```

**Reply by adamcameron** [reply]
Cool: good to know! That would have solved my issue probably, but have just realised the internal payload is JSON but the content-type I need to return is `text/event-stream`, which `provides` doesn't permit :-/

Any reason the implementation of `provides` specifically expects a limited subset of CFWheels-specific "shortcodes" ("json", "html" etc), not just... _a mime type_? They're kinda industry-standard, yeah? Seems like CFWheels is going out of its way to hamstring its own functionality, but I'm obvs missing something. I have to admit I cannot be arsed wading through the code to work out why it's doing what it's doing.

Cheers for the info though, Tom. As usual you are gold.

**Reply by neokoenig** [reply]
You can add your own. Chuck this (or equivalent) in config/settings
`addFormat(extension="jsonish", mimeType="text/event-stream")`

then you can use provides(`jsonish") etc...

**Comment by adamcameron** [comment]
Nice one. That worked.
